src/Sarif.Viewer.VisualStudio/Controls/InfoBar.cs

```diff
@@ -197,5 +223,21 @@ public void OnActionItemClicked(IVsInfoBarUIElement infoBarUIElement, IVsInfoBar
                 this.clickAction?.Invoke(actionItem);
             }
         }
+
+        private static async Task AddInfoBarIfRequiredAsync(
+            ExceptionalConditions detectedConditions,
+            ExceptionalConditions individualCondition,
+            string message,
+            ImageMoniker imageMoniker)
+        {
+            // It's safe to access and update the dictionary because this method is called inside
+            // the write lock.
+            if ((detectedConditions & individualCondition) == individualCondition && !s_infoBarToConditionDictionary.Values.Contains(individualCondition))
+            {
+                var infoBar = new InfoBar(message, imageMoniker: imageMoniker);
+                await infoBar.ShowAsync();
+                s_infoBarToConditionDictionary.Add(infoBar, individualCondition);
```

Comment:
Refer this [link ](https://docs.microsoft.com/en-us/dotnet/standard/collections/thread-safe/how-to-add-and-remove-items)for ConcurrentDictionary<>, TryAdd() is prefered:
if ( !s_infoBarToConditionDictionary.Add(xxxx, xxxx))
{
      throw new Exception("");
}

Reply:
Fixed, thanks.

---
In reply to: [527332697](https://github.com/microsoft/sarif-visualstudio-extension/pull/278#discussion_r527332697) [](ancestors = 527332697)